### PR TITLE
Update README to follow Heroku best practices for buildpacks

### LINKED ISSUE #229: Update usage instructions to no longer pin buildpack version
Hi!

As mentioned in #228, we recently had a customer open a support ticket who encountered the Heroku-20 error in #216, since they were pinned to the older v.1.17 buildpack release, which doesn't have Heroku-20 support.

Heroku buildpacks are something that most users tend to "set and forget", so whilst in general pinning versions of a thing would be recommended, for buildpacks we recommend against it, otherwise it makes it very hard to push out urgent fixes. This obviously means the buildpack needs to be more mindful of breaking changes (eg either avoiding them, or using a combination of warnings-in-advance, opt-in changes, and clear error messaging when breaking changes are made), but it's normally worth the trade-off.

As such, would it be possible to update the docs here, to remove the version pinning?
https://github.com/DataDog/heroku-buildpack-datadog/blame/2a5b7d7e62185a62c38c7bf6fd86e25fb172956e/README.md#L22

The current pinning usage example was added in #120, since at the time the Heroku docs did suggest pinning, however they've seen been updated to recommend only pinning for debugging:
https://devcenter.heroku.com/articles/buildpacks#buildpack-references

Advanced users can still choose to pin should they wish (plus any customers who encounter a breaking change can always pin retrospectively).

Many thanks :-)

### PR DESCRIPTION
**Pull request by arapulido**
Update install instructions to recommend not pinning the buildpack. Fixes #229

Remove alert-warning div, as it wasn't being rendered correctly in the docs site

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,29 +19,38 @@ heroku buildpacks:add heroku/ruby
...
 ```
 
 Replace `<DATADOG_API_KEY>` with your [Datadog API key][6].
-Replace `<DATADOG_BUILDPACK_RELEASE>` with the [Buildpack release][7] you want to use.
 
 Once complete, the Datadog Agent is started automatically when each dyno starts.
 
 The Datadog Agent provides a listening port on `8125` for statsd/dogstatsd metrics and events. Traces are collected on port `8126`.
 
-<div class="alert alert-warning">
```
Comment (jeremy-lq) [inline README.md:36]: Can use standard links if we want to retain the warning. No strong preference.

File: README.md [hunk 2]
```diff
@@ -19,29 +19,38 @@ heroku buildpacks:add heroku/ruby
...
+
+Heroku recommends to always use the latest commit of a buildpack. If you need to pin the buildpack version, you can do so by specifying the buildpack release tag:
+
+```
+heroku buildpacks:add --index 1 https://github.com/DataDog/heroku-buildpack-datadog.git#<DATADOG_BUILDPACK_RELEASE>
+```
+
+Replace `<DATADOG_BUILDPACK_RELEASE>` with the [Buildpack release][7] you want to use.
+
+In both cases, the version of the Datadog agent can be pinned as well by setting `DD_AGENT_VERSION` environment variable.
```
Comment (jeremy-lq) [inline README.md:53]: Suggest:

By default, the buildpack pins the latest version of the Datadog Agent at the time of release. You can pin the Agent to an earlier version by setting the `DD_AGENT_VERSION` environment variable.

**Review by jeremy-lq — CHANGES_REQUESTED** [review]

**Review by jeremy-lq — APPROVED** [review]

**Comment by agrobbin** [comment]
@arapulido does this mean that an upgrade to the buildpack version (and by extension the agent version) no longer requires a purging of the Heroku build cache?

**Comment by arapulido** [comment]
@agrobbin Good point. This is still necessary, but I understand that if you are using master, then it is more difficult to know when this is coming. I will work on this to make it automated when the agent version changes.

**Pull request merged**